internal/wrapper/checkargs.go

```diff
@@ -0,0 +1,71 @@
+package wrapper
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/Masterminds/semver/v3"
+)
+
+func checkStateCommand(args []string, version *semver.Version) error {
+	versionImport, _ := semver.NewConstraint(">= 1.5.0")
+	versionMoved, _ := semver.NewConstraint(">= 1.1.0")
+	versionRemoved, _ := semver.NewConstraint(">= 1.7.0")
+	STATE_COMMAND_VAR := "TF_DEMUX_ALLOW_STATE_COMMANDS"
+
+	errorMsg := func(command string, suggestion string) error {
+		return fmt.Errorf("refusing to execute '%s' command - use a '%s' configuration block instead, or set %s=true.", command, suggestion, STATE_COMMAND_VAR)
+	}
+
```

Comment:
one last suggestion - if you check the allowStateCommand here, if its true you can just return early and not include the check in the following conditionals